Assets/Scripts/Controllers/PlayerController.cs

```diff
@@ -45,7 +52,14 @@ protected new void Update()
             doneTurn = !Action();
             if (doneTurn)
             {
-                GameLoopManager.SetPlayerTurn(false);
+                if (healthBar.health > 0)
+                {
+                    GameLoopManager.SetPlayerTurn(false);
+                }
+                else
+                {
+                    Die();
```

Comment:
Why handle death call here as well? We already handle this logic here:

https://github.com/AndreiBaetica/Dungeoneer/blob/master/Assets/Scripts/Controllers/CharController.cs#L217-L221